contrib/sly-package-fu.el

```diff
@@ -317,4 +332,126 @@ symbol in the Lisp image if possible."
 
 (defalias 'sly-export-structure 'sly-export-class)
 
+;; 
+;; Dealing with import-from
+;;
+;; MELPA version: (buffer-substring-no-properties point1 pount2)
+;; ~/.emacs.d/elpa/sly-20171111.1604/contrib/sly-package-fu.el
+
+(defun sly-search-import-from (package)
+  ;; Suppose, we are in the defpackage sexp
+  (let* ((normalized-package (sly-package-normalize-name package))
+         (regexp (format "(:import-from[ \t']*\\(:\\|#:\\)?%s"
+                         (regexp-quote (regexp-quote normalized-package))))
+         (search-result (re-search-forward regexp nil t)))
+    (message "Normalized: %s, regex: %s" normalized-package
+             regexp)
+    (when search-result
+      ;; import-from clause was found
+      t)))
+
+
+(defun sly-package-write-imported (symbol)
+  "Like rites given symbol to the end of the :import-from clause.
+
+   Expects that current point is right after the last symbol
+   in the :import-from sexp:
+
+   (:import-from #:weblocks/requests
+                 #:get-parameters<point>)"
+
+  (goto-char point)
+  (down-list)
+  (sly-end-of-list))
+
+
+(defun get-piece (chars)
+  (buffer-substring-no-properties (point)
+                                  (+ (point)
+                                     chars)))
+
+(defun sly-package-create-new-import-from (package symbol)
+  (sly-goto-package-source-definition (sly-current-package))
+  (forward-sexp)
+  ;; Now, search last :import-from or :use form
+  (cond
+    ((re-search-backward "(:\\(use\\|import-from\\)" nil t)
+     ;; Skip found expression:
+     (forward-sexp)
+     ;; and insert a new (:import-from <package> <symbol>) form.
+     (newline-and-indent)
+     (let ((symbol-name (sly-format-symbol-for-defpackage symbol))
+           (package-name (sly-format-symbol-for-defpackage package)))
+       (insert "(:import-from )")
+       (backward-char)
+       (insert package-name)
+       (newline-and-indent)
+       (insert symbol-name)))
+    (t (error "Unable to find :use form in the defpackage form."))))
+
+
+(defun sly-import-symbol-from (symbol)
+  "Accepts "
+  (save-excursion
+   ;; First, will go to the package definition
+   (sly-goto-package-source-definition (sly-current-package))
+
+   (let* ((package (funcall sly-import-symbol-package-transform-function
+                            (sly-cl-symbol-package symbol)))
+          (simple-symbol (sly-cl-symbol-name symbol))
+          (import-exists (when package
+                           (sly-search-import-from package))))
+
+     ;; We only process symbols in fully qualified form like
+     ;; weblocks/request:get-parameter
+     (when package
```

Comment:
Perhaps its useful, instead of `when package` to issue an `error` or an `user-error`. Like
```lisp
(unless package
  (user-error "Can't guess the package designator of %s" symbol))
```

Reply:
Done.